agent/exec/executor.go

```diff
@@ -15,9 +15,16 @@ type Executor interface {
 	Configure(ctx context.Context, node *api.Node) error
 
 	// Controller provides a controller for the given task.
-	Controller(t *api.Task) (Controller, error)
+	Controller(t *api.Task, secrets SecretStore) (Controller, error)
 
 	// SetNetworkBootstrapKeys passes the symmetric keys from the
 	// manager to the executor.
 	SetNetworkBootstrapKeys([]*api.EncryptionKey) error
 }
+
+// SecretStore contains secret data necessary for the Controller.
+type SecretStore interface {
```

Comment:
I might have gone with `SecretProvider` if this is meant to be a read-only interface. But TBH I think `SecretStore` is fine.